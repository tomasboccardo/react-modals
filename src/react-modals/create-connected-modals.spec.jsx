import React from 'react';
import createConnectedModals, { withActionDispatching } from './create-connected-modals';

describe('createConnectedModals', () => {
  let connectorMock;

  beforeEach(() => {
    connectorMock = jest.fn();
  });

  it('should call the connector', () => {
    createConnectedModals(connectorMock);
    expect(connectorMock).toHaveBeenCalled();
  });
});

describe('withActionDispatching', () => {
  let props;
  let ModalsMock;
  let withActionDispatchingModals;

  beforeEach(() => {
    props = {
      modalsConfig: [{
        title: 'title',
        buttons: [],
        onButtonClick: jest.fn(),
      }],
      dispatch: jest.fn(),
    };
    ModalsMock = () => 'Modals';
    const WithActionDispatchingModals = withActionDispatching(ModalsMock);
    withActionDispatchingModals = shallow(<WithActionDispatchingModals {...props} />);
  });

  it('should match the snapshot', () => {
    expect(withActionDispatchingModals).toMatchSnapshot();
  });

  it('should dispatch actions', () => {
    withActionDispatchingModals.props().modalsConfig[0].onButtonClick({
      action: {
        type: 'action',
      },
    });

    expect(props.dispatch).toHaveBeenCalled();
  });

  it('should support buttons with no actions', () => {
    withActionDispatchingModals.props().modalsConfig[0].onButtonClick({});

    expect(props.modalsConfig[0].onButtonClick).toHaveBeenCalled();
  });
});
