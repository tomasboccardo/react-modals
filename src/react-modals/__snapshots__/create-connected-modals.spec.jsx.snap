// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActionDispatching should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DispatcherModals
    dispatch={[Function]}
    modalsConfig={
        Array [
            Object {
              "buttons": Array [],
              "onButtonClick": [Function],
              "title": "title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "modalsConfig": Array [
        Object {
          "buttons": Array [],
          "onButtonClick": [Function],
          "shouldShow": true,
          "title": "title",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "modalsConfig": Array [
          Object {
            "buttons": Array [],
            "onButtonClick": [Function],
            "shouldShow": true,
            "title": "title",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
